### DISCUSSION TITLE
Adonis Encryption  react js decryption and vice a versa

### DISCUSSION THREAD

**Discussion by pawan-khade** [adonisjs/core#3779] [General]
I want to encrypt message in adonis 5 using aes 256 algorithm with IV. using that iv I want to decrypt that message in react js   and vice a versa how to do that? I have searched very much but nothing is working

**Comment by pawan-khade** [comment]
This is my Node JS Code for encryption and decryption

```
public async encrypt(message)
  {
    let key = Env.get('ID1');
    const iv = crypto.randomBytes(16).toString("hex").slice(0, 16);
    const encrypter = crypto.createCipheriv("aes-256-cbc", key, iv);
    let encryptedMsg = encrypter.update(message, 'utf8', 'hex');
    encryptedMsg += encrypter.final("hex");

    return iv+'-'+encryptedMsg;
  }

  public async decrypt(encryptedMsg)
  {
    let key = Env.get('ID1');
    let iv = encryptedMsg.split('-')[0];
    let message = encryptedMsg.split('-')[1];

    const decrypter = crypto.createDecipheriv("aes-256-cbc", key, iv);
    let decryptedMsg = decrypter.update(message, 'hex', 'utf8');
    decryptedMsg += decrypter.final("utf8");
  }

```

How to write its reverse code in react js.

**Comment by thetutlage** [comment]
Nope, the `iv` is not return during the encrypt call. Also, I am wondering, if encryption has any use when you will share the `iv` and the `key` to decrypt the value on client side

**Reply by pawan-khade** [reply]
During encryption I am only returning IV with encrypted payload not a key. so that same IV will be used to do decryption along with secret key on react side

**Reply by thetutlage** [reply]
But its going to be the same key right?

**Reply by pawan-khade** [reply]
secret key will be different and iv will be different

**Reply by pawan-khade** [reply]
const iv = crypto.randomBytes(16).toString("hex").slice(0, 16);
is generating IV dynamically

**Reply by pawan-khade** [reply]
and yes backend and front end will be having same secret key

**Reply by thetutlage** [reply]
To be honest, I don't properly understand the use case here and not sure if changing the return type of the `encrypt` method because it will lead to a breaking change.

Maybe, the best option is to roll out your own internal service for Encryption and decryption.

**Reply by pawan-khade** [reply]
Thanks for timely reply. I will surely do that.

**Reply by darshithedpara** [reply]
@pawan-khade Did you find any solution for this?
